apps/rush-lib/src/cli/test/RushCommandLineParser.test.ts

```diff
@@ -279,4 +282,63 @@ describe('RushCommandLineParser', () => {
       });
     });
   });
-});
\ No newline at end of file
+
+  describe(`in repo with tests for add`, () => {
+    describe(`'add' action`, () => {
+      it(`adds a dependency to just one repo in the workspace`, () => {
+        const startPath: string = resolve(__dirname, 'addRepo');
+        const aPath: string = resolve(__dirname, 'addRepo/a');
+        FileSystem.deleteFile(resolve(__dirname, `addRepo/a/package-deps.json`));
+
+        // Create a Rush CLI instance. This instance is heavy-weight and relies on setting process.exit
+        // to exit and clear the Rush file lock. So running multiple `it` or `describe` test blocks over the same test
+        // repo will fail due to contention over the same lock which is kept until the test runner process
+        // ends.
+        const parser: RushCommandLineParser = new RushCommandLineParser({ cwd: startPath });
+
+        // Switching to the "a" package of addRepo
+        process.chdir(aPath);
+
+        // Mock the command
+        process.argv = ['pretend-this-is-node.exe', 'pretend-this-is-rush', 'add', '-p', 'assert'];
+
+        expect.assertions(2);
+        return expect(parser.execute()).resolves.toEqual(true)
```

Comment:
I think this is too high in the stack. The test should only call code directly related to updating the `package.json`s, and it should stub out the code that runs the package installation and modifies the files on disk.

If you think that's too big a change for this PR, let's check this in without unit tests for now.

Reply:
I agree with you! I did the tests this way since I'm struggling a bit to dive in the code to understand what needs to be stubbed. Do we have examples on other tests? Any form of guidance is welcome!

I will resolve the other pieces of feedback today. I will have time to work on the logic changes next week!

Thank you for your time 🌞